server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationState.java

```diff
@@ -34,7 +35,9 @@
  * The core class of the cluster state coordination algorithm, directly implementing the
  * <a href="https://github.com/elastic/elasticsearch-formal-models/blob/master/ZenWithTerms/tla/ZenWithTerms.tla">formal model</a>
  */
-public class CoordinationState extends AbstractComponent {
+public class CoordinationState {
+
+    private final Logger logger = LogManager.getLogger(getClass());
```

Comment:
if it's a private logger to the class, this should probably not dynamically resolve to the instance class type but be `CoordinationState.class`. You can then also make the logger static here. Same for the other occurrences

Reply:
Makes sense, thanks.